core/primitives/src/types.rs

```diff
@@ -890,8 +890,8 @@ pub enum TransactionOrReceiptId {
 
 /// Cache for compiled modules
 pub trait CompiledContractCache: Send + Sync {
-    fn put(&self, key: &[u8], value: &[u8]) -> Result<(), std::io::Error>;
-    fn get(&self, key: &[u8]) -> Result<Option<Vec<u8>>, std::io::Error>;
+    fn put(&self, key: &CryptoHash, value: Vec<u8>) -> Result<(), std::io::Error>;
+    fn get(&self, key: &CryptoHash) -> Result<Option<Vec<u8>>, std::io::Error>;
```

Comment:
Arguably, the old signatures were better: it's nobody's business what *exactly* contract-runtime is storing in the cache. They types of `key` and `value` might be private to contract runtime. But it also not a hard thing to change down the line, if we decide that we want to have more complicated key structure, and even that seems unlikely.